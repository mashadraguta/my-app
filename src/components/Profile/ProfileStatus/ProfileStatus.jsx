

import React, { Component } from 'react';
import s from './ProfileStatus.module.css'


class ProfileStatus extends Component {

    state = {
        editMode: false,
        userStatus: this.props.userStatus,
    }


    activateEditMode = () => {
  
        this.setState({
            editMode: true,
        })


    }

    deactivateEditMode = () => {

        this.setState({
            editMode: false,
        })
        this.props.updateStatusThunkCreator(this.state.userStatus)
    }

    onStatusChange = (e) => {
        this.setState({
            userStatus: e.currentTarget.value,
        })


    }

    componentDidUpdate = (prevProps, prevState) => {

        if (prevProps.userStatus !== this.props.userStatus) {
            this.setState({
                userStatus: this.props.userStatus
            })
        }
    }
    render() {

        return (
            <div className={s.wrapper}>
                {!this.state.editMode &&
                    <div>
                        <span onDoubleClick={this.activateEditMode}>
                            {this.props.userStatus || "no more crying in the rain"}
                        </span>
                    </div>
                }
                {this.state.editMode &&

                    <div><input onBlur={this.deactivateEditMode}
                        autoFocus={true}
                        onChange={this.onStatusChange}
                        value={this.state.userStatus}></input></div>
                }

            </div>

        );
    }
}

export default ProfileStatus;
