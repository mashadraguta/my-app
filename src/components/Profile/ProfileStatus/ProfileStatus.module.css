.wrapper {
    padding: 20px;
}



.statusSpan {

    border: 2px solid black;
    width: 100%;
    height: 100%;
    text-align: center;
    position: relative;
    padding: 50px;
}


.statusCont .spanHoverComp {
    visibility: hidden;
}

.statusHover:hover {
    display: inline-block;
    width: 100%;
    height: 100%;
    visibility: hidden;
}

.statusHover {

    min-width: 100%;
    min-height: 100%;

}

.statusCont:hover .spanHoverComp {
    visibility: visible;
}

.container {

    width: 100%;
    height: 100%;
    text-align: center;
    position: absolute;
    padding: 50px;


}

.container:hover {
    width: 100%;
    height: 100%;
    background-color: #cd853f;

    transition: background-color 0.2s cubic-bezier(0.3, 0, 0.5, 1);
    /* transition: 0.3s ease; */
    cursor: pointer;
    top: 0;
    left: 0;
}

.spanHoverComp {
    color: #faf0be;
}

.statusInput {
    border: 2px none black;
    width: 100%;
    height: 100%;
    text-align: center;

    padding: 50px;
}

.statusDiv {
    padding: 30px 30px 30px 0;
    font-style: italic;
}