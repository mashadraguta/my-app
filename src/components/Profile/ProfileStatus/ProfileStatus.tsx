

import React, { ChangeEvent, Component } from 'react';
import s from './ProfileStatus.module.css'


type PropsType = {
    userStatus: string
    updateStatusThunkCreator: (userStatus: string) => void
}

type StateType = {
    editMode: boolean
    userStatus: string
}

class ProfileStatus extends Component<PropsType, StateType> {

    state = {
        editMode: false,
        userStatus: this.props.userStatus,
    }


    activateEditMode = () => {

        this.setState({
            editMode: true,
        })


    }

    deactivateEditMode = () => {

        this.setState({
            editMode: false,
        })
        this.props.updateStatusThunkCreator(this.state.userStatus)
    }

    onStatusChange = (e: ChangeEvent<HTMLInputElement>) => {
        this.setState({
            userStatus: e.currentTarget.value,
        })


    }

    componentDidUpdate = (prevProps: PropsType, prevState: StateType) => {

        if (prevProps.userStatus !== this.props.userStatus) {
            this.setState({
                userStatus: this.props.userStatus
            })
        }
    }
    render() {

        return (
            <div className={s.wrapper}>
                {!this.state.editMode &&
                    <div>
                        <span onDoubleClick={this.activateEditMode}>
                            {this.props.userStatus || "no more crying in the rain"}
                        </span>
                    </div>
                }
                {this.state.editMode &&

                    <div><input onBlur={this.deactivateEditMode}
                        autoFocus={true}
                        onChange={this.onStatusChange}
                        value={this.state.userStatus}></input></div>
                }

            </div>

        );
    }
}

export default ProfileStatus;
