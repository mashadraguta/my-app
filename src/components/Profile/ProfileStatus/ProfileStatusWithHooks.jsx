

import React, { useState, useEffect } from 'react';
import s from './ProfileStatus.module.css'


const ProfileStatusWithHooks = (props) => {

    let [editMode, setEditMode] = useState(false);
    let [userStatus, setStatus] = useState(props.userStatus);

    useEffect(() => {

        setStatus(props.userStatus)

    }, [props.userStatus])

    const activateEditMode = () => {
        setEditMode(true)
    }
    const deactivateEditMode = () => {
        setEditMode(false);
        props.updateStatusThunkCreator(userStatus);
    }
    const onStatusChange = (e) => {
        setStatus(e.currentTarget.value);
    }


    return (
        <div className={s.wrapper}>
            {!editMode &&
                <div>
                    <span onDoubleClick={activateEditMode} >
                        {props.userStatus || "no more crying in the rain"}
                    </span>
                </div>
            }
            {editMode &&
                <div><input onBlur={deactivateEditMode}
                    onChange={onStatusChange}
                    value={userStatus}></input></div>
            }

        </div>

    );
}


export default ProfileStatusWithHooks;
