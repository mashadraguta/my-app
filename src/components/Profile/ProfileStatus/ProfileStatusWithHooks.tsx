

import React, { useState, useEffect } from 'react';
import s from './ProfileStatus.module.css'




const SpanHover = () => {
    return (
        <div>
            <div className={s.spanHoverComp}>doubleclick to change status</div>
        </div>
    );
}


type MapDispachToProps = {
    updateStatusThunkCreator: (userStatus: string | undefined) => void

}
type MapStateToProps = {
    userStatus: string | undefined

}



type PropsType = MapDispachToProps & MapStateToProps


const ProfileStatusWithHooks: React.FC<PropsType> = (props) => {

    let [editMode, setEditMode] = useState(false);
    let [userStatus, setStatus] = useState(props.userStatus);

    useEffect(() => {

        setStatus(props.userStatus)

    }, [props.userStatus])

    const activateEditMode = () => {
        setEditMode(true)
    }
    const deactivateEditMode = () => {
        setEditMode(false);
        props.updateStatusThunkCreator(userStatus);
    }
    const onStatusChange = (e: { currentTarget: { value: React.SetStateAction<string | undefined> } }) => {
        setStatus(e.currentTarget.value);
    }


    return (
        <div className={s.wrapper}>
            {!editMode &&
                <div className={s.statusCont}>
                    <div className={s.statusDiv}>Hello! How do you feel today?</div>
                    <div onDoubleClick={activateEditMode} className={s.statusSpan}>
                        <div className={s.container}>
                            <span className={s.spanHoverComp}>doubleclick to change status</span>
                        </div>
                        <div className={s.statusHover}>{props.userStatus || "no more crying in the rain"}</div>
                    </div>

                </div>
            }
            {editMode &&
                <div>
                    <div className={s.statusDiv}>Hello! How do you feel today?</div>
                    <input onBlur={deactivateEditMode}
                        className={s.statusInput}
                        onChange={onStatusChange}
                        value={userStatus}></input></div>
            }

        </div>

    );
}


export default ProfileStatusWithHooks;
