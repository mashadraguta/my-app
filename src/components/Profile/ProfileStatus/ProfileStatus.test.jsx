
import React from "react";
import { create } from "react-test-renderer";
import ProfileStatus from "./ProfileStatus";

describe("ProfileStatus component", () => {
    test("status from props should be in the state", () => {

        const component = create(<ProfileStatus userStatus="SUBSCRIBE" />);
        const instance = component.getInstance();
        expect(instance.state.userStatus).toBe("SUBSCRIBE");
    });
    test("span should be displayed at doubleClick", () => {
        const component = create(<ProfileStatus />);
        const root = component.root;
        const span = root.findByType("span");
        expect(span.length).not.toBeNull();
    });
    test("after creation <input> shouldn't be displayed", () => {
        const component = create(<ProfileStatus userStatus="it-kamasutra.com" />);
        const root = component.root;
        expect(() => {
            let input = root.toBeNull("input");
        }).toThrow();
    });
});