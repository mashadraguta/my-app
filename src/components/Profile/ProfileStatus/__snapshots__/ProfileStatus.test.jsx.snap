// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileStatus component status from props should be in the state 1`] = `
<div
  className="wrapper"
>
  <div>
    <span
      onDoubleClick={[Function]}
    >
      no more crying in the rain
    </span>
  </div>
</div>
`;
